stackslib/src/chainstate/stacks/transaction.rs

```diff
@@ -138,52 +138,89 @@ fn ClarityVersion_consensus_deserialize<R: Read>(
     }
 }
 
+impl StacksMessageCodec for TenureChangeCause {
+    fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), codec_error> {
+        let byte = (*self) as u8;
+        write_next(fd, &byte)
+    }
+
+    fn consensus_deserialize<R: Read>(fd: &mut R) -> Result<TenureChangeCause, codec_error> {
+        let byte: u8 = read_next(fd)?;
+        TenureChangeCause::try_from(byte).map_err(|_| {
+            codec_error::DeserializeError(format!("Unrecognized TenureChangeCause byte {byte}"))
+        })
+    }
+}
+
+impl StacksMessageCodec for TenureChangePayload {
+    fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), codec_error> {
+        write_next(fd, &self.previous_tenure_end)?;
+        write_next(fd, &self.previous_tenure_blocks)?;
+        write_next(fd, &self.cause)?;
+        write_next(fd, &self.pubkey_hash)?;
+        write_next(fd, &self.signers)
+    }
+
+    fn consensus_deserialize<R: Read>(fd: &mut R) -> Result<TenureChangePayload, codec_error> {
+        Ok(TenureChangePayload {
+            previous_tenure_end: read_next(fd)?,
+            previous_tenure_blocks: read_next(fd)?,
+            cause: read_next(fd)?,
+            pubkey_hash: read_next(fd)?,
+            signers: read_next(fd)?,
+        })
+    }
+}
+
 impl StacksMessageCodec for TransactionPayload {
     fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), codec_error> {
-        match *self {
-            TransactionPayload::TokenTransfer(ref address, ref amount, ref memo) => {
+        match self {
+            TransactionPayload::TokenTransfer(address, amount, memo) => {
                 write_next(fd, &(TransactionPayloadID::TokenTransfer as u8))?;
                 write_next(fd, address)?;
                 write_next(fd, amount)?;
-                write_next(fd, memo)?;
```

Comment:
I wouldn't consider these additional exit points, those are created with `return`. The only exit point here is when the output of the `match` expression is returned (unless you count each `?`, in which case there are now fewer)

I can change it back if you want, it's not that important, but IMO it's cleaner and less code to return `Result` types directly when possible, like:

```rust
fn a_function() -> Result<(), Error> {
    // ...
    another_function()
}
```

instead of

```rust
fn a_function() -> Result<(), Error> {
    // ...
    another_function()?;
    Ok(())
}
```

Reply:
The issue here is that each arm of the `match` expression is now an exit point, so reasoning about where execution returned is a bit harder.  Before, there was only a single exit point.

Reply:
No problem, I can change it back